File: Chathub.API/Chathub.API/Domain/Data/Models/ChatRoom.cs
namespace Chathub.API.Domain.Data.Models
{
    public class ChatRoom
    {
        public Guid Id { get; set; }
        public string Name { get; set; }
        public string Description { get; set; }
        public bool CanShareFile { get; set; }
        public bool CanManageFile { get; set; }
        public bool CanManageMember { get; set; }
        public bool CanGrantRight { get; set; }
        public List<ChatMember> MemberList { get; set; }
        public List<ChatMessage> MessageList { get; set; }
        public List<ChatFile> FileList { get; set; }
    }
}

File: Chathub.UI/src/models/ChatRoom.ts
import ChatMember from "./ChatMember";
import ChatMessage from "./ChatMessage";
import ChatFile from "./ChatFile";

type ChatRoom = {
    id: string,
    name: string,
    description: string,
    canShareFile: boolean,
    canManageFile: boolean,
    canManageMember: boolean,
    canGrantRight: boolean,
    memberList: ChatMember[],
    contentList: (ChatMessage|ChatFile)[]
}

export default ChatRoom;